sentry_sdk/integrations/aws_lambda.py

```diff
@@ -37,14 +78,31 @@ def sentry_handler(event, context, *args, **kwargs):
 
         # If an integration is there, a client has to be there.
         client = hub.client  # type: Any
+        configured_time_in_millis = context.get_remaining_time_in_millis()
 
         with hub.push_scope() as scope:
             with capture_internal_exceptions():
                 scope.clear_breadcrumbs()
                 scope.transaction = context.function_name
-                scope.add_event_processor(_make_request_event_processor(event, context))
+                scope.add_event_processor(_make_request_event_processor(event, context, configured_time_in_millis))
 
             try:
+                # Checking if parameter to check timeout is set True
+                if integration.get_check_timeout_error():
+                    # Starting the Timeout thread only if the configured time is greater than Timeout threshold value
+                    if configured_time_in_millis > TIMEOUT_THRESHOLD_MILLIS:
+                        remaining_time_in_sec = (configured_time_in_millis - TIMEOUT_THRESHOLD_MILLIS)/SECONDS_CONVERSION_FACTOR
+
+                        configured_time_in_sec = configured_time_in_millis / SECONDS_CONVERSION_FACTOR
+                        configured_time = int(configured_time_in_sec)
+
+                        # Setting up the exact integer value of configured time(in seconds)
+                        if configured_time < configured_time_in_sec:
+                            configured_time = configured_time + 1
```

Comment:
`time.sleep` can work with floating point numbers, this dance to round up to the next integer is not really needed, is it?

We could also probably get rid of `SECONDS_CONVERSION_FACTOR` and simply use subsecond timeout in the tests, perhaps?